comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/readers/DeleteEmptyDirectoriesReader.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * ComiXed - A digital comic book library management application.
+ * Copyright (C) 2022, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.batch.comicbooks.readers;
+
+import static org.comixedproject.service.admin.ConfigurationService.CFG_LIBRARY_DELETE_EMPTY_DIRECTORIES;
+import static org.comixedproject.service.admin.ConfigurationService.CFG_LIBRARY_ROOT_DIRECTORY;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.service.admin.ConfigurationService;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.NonTransientResourceException;
+import org.springframework.batch.item.ParseException;
+import org.springframework.batch.item.UnexpectedInputException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+/**
+ * <code>DeleteEmptyDirectoriesReader</code> loads all directories below the library root that
+ * contain no contents.
+ *
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class DeleteEmptyDirectoriesReader implements ItemReader<File> {
+  @Autowired private ConfigurationService configurationService;
+
+  List<File> directories;
+
+  @Override
+  public File read()
+      throws Exception, UnexpectedInputException, ParseException, NonTransientResourceException {
```

Comment:
If we throws Exception, we don't need to throw UnexpectedInputException, ParseException, NonTransientResourceException ?

Reply:
True. I'll clean that up as well.